lib/kits/core-ui/components/paged-publication/shopping-list.js

```diff
@@ -0,0 +1,285 @@
+            translations,
+            offers: transformSavedOffers(clientStorage)?.filter(
+                (offer) => !offer.is_ticked
+            ),
+            tickedOffers: transformSavedOffers(clientStorage)?.filter(
+                (offer) => offer.is_ticked
+            ),
+            hasTicked:
+                transformSavedOffers(clientStorage)?.filter(
+                    (offer) => offer.is_ticked
+                ).length > 0
+        });
+
+        addEventListeners();
+        e.stopPropagation();
+    };
+
+    const addClearListListener = () => {
+        const clearBtn = container.querySelector(
+            '.sgn-shopping-clear-list-btn'
+        );
+
+        clearBtn?.addEventListener('click', (e) => {
+            e.stopPropagation();
+            clientLocalStorage.set('paged-publication-saved-offers', []);
+            container.innerHTML = Mustache.render(template, {
+                translations,
+                offers: []
+            });
+            addEventListeners();
+        });
+    };
+
+    const addClearTickedListListener = () => {
+        const clearTickedBtn = container.querySelector(
+            '.sgn-shopping-clear-ticked-list-btn'
+        );
+        const clientStorage = clientLocalStorage.get(
+            'paged-publication-saved-offers'
+        );
+        const validOffers = clientStorage?.filter((offer) => !offer.is_ticked);
+
+        clearTickedBtn?.addEventListener('click', (e) => {
+            e.stopPropagation();
+            clientLocalStorage.set(
+                'paged-publication-saved-offers',
+                validOffers
+            );
+            container.innerHTML = Mustache.render(template, {
+                translations,
+                offers: transformSavedOffers(validOffers)
+            });
+            addEventListeners();
+        });
+    };
+
+    const addPrintListener = () => {
+        const printBtn = container.querySelector(
+            '.sgn-shopping-print-list-btn'
+        );
+
+        printBtn?.addEventListener('click', () => {
+            window.print();
+        });
+    };
+
+    const addShareListener = () => {
+        const shareBtn = container.querySelector(
+            '.sgn-shopping-share-list-btn'
+        );
+        const clientStorage = clientLocalStorage.get(
+            'paged-publication-saved-offers'
+        );
+        const textData = formatListToShare(transformSavedOffers(clientStorage));
+
+        const shareData = {
+            title: 'My Shopping List',
+            text: textData,
+            url: document.location.href
+        };
+
+        shareBtn?.addEventListener('click', () => {
+            if (navigator.share) {
+                navigator
+                    .share(shareData)
+                    .then(() => {
+                        console.log('Shared data:', shareData);
+                    })
+                    .catch(console.error);
+            }
```

Comment:
Firefox and Safari like it if you ask `navigator.canShare(shareData)` first, but many other browsers don't have it, so you need to check if it exists and ask first if it does but not if it doesn't